core/src/main/scala/cats/data/IndexedStateT.scala

```diff
@@ -223,6 +223,11 @@ object IndexedStateT extends IndexedStateTInstances with CommonStateTConstructor
     IndexedStateT(_ => F.map(fsb)(s => (s, ())))
 }
 
+private[data] trait CommonStateTConstructors0 extends CommonStateTConstructors {
```

Comment:
Is there any particular reason to introduce this new trait as opposed to just putting this method directly into the `object`?

Reply:
Binary compatibility: https://github.com/typelevel/cats/pull/2533#issuecomment-424601003

Reply:
To be precise, the function is not only provided on the `IndexedStateT` object, but also on `StateT`. This is the reason why it has to go into a trait in the first place.

Reply:
Oh okay that makes sense. Thanks for the explanation.